app/cni/pkg/cni/kubernetes_linux.go

```diff
@@ -2,41 +2,108 @@ package cni
 
 import (
 	"context"
+	"slices"
 
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-retry"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/metadata"
+	k8s_util "github.com/kumahq/kuma/pkg/plugins/runtime/k8s/util"
 )
 
-func newKubeClient(conf PluginConf) (*kubernetes.Clientset, error) {
-	kubeconfig := conf.Kubernetes.Kubeconfig
+func newKubeClient(logger logr.Logger, conf PluginConf) (*kubernetes.Clientset, error) {
+	logger = logger.WithValues("kubeconfig", conf.Kubernetes.Kubeconfig)
+
 	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfig},
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: conf.Kubernetes.Kubeconfig},
 		&clientcmd.ConfigOverrides{},
 	).ClientConfig()
 	if err != nil {
-		log.Error(err, "failed setting up kubernetes client with kubeconfig", "kubeconfig", kubeconfig)
+		logger.Error(err, "failed setting up kubernetes client with kubeconfig")
 		return nil, err
 	}
 
-	log.V(1).Info("set up kubernetes client with kubeconfig", "kubeconfig", kubeconfig, "config", config)
+	logger.V(1).Info("set up kubernetes client with kubeconfig", "config", config)
 
 	return kubernetes.NewForConfig(config)
 }
 
-// getK8sPodInfo returns information of a POD
-func getKubePodInfo(ctx context.Context, client *kubernetes.Clientset, podName, podNamespace string) (int, map[string]struct{}, map[string]string, error) {
-	pod, err := client.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
-	log.V(1).Info("pod info", "pod", pod)
+func getAndValidatePodAnnotations(
+	ctx context.Context,
+	logger logr.Logger,
+	conf *PluginConf,
+	k8sArgs K8sArgs,
+) (map[string]string, bool, error) {
```

Comment:
After reading the whole func and usage as well, I'm confused what this bool means. Could you explain it?

Reply:
Added comment